homework-submission/sobhiweek2/src/App.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import './App.css';
+import todos from "./source/todos.json";
+import Header from "./source/Header";
+import Description from "./source/Description";
+import DeadLine from "./source/DeadLine";
+    
+
+export default class App extends React.Component {
+
+  state = { todos };
+
+  handelCheck = () => { this.setState({ todos: this.state.todos })};
+
+  render() {
+
+    const todosState = this.state.todos;
+
+    const todosRender = todosState.map((elem) =>
+
+      <div key={elem.id}>
+       <span style={elem.done ? { textDecoration: 'line-through' } : { }}> 
+        <Description description={elem.description} />
+        <DeadLine deadline={elem.deadline} />
+        <input type="checkbox" className="checkbox"
+         onClick={this.handelCheck.bind(null, elem)}
```

Comment:
You are binding the `handleClick` with *elem* however you are not using any arguments in the function above. (line 13).  Also, you don't need to use *bind* in this case, instead call the function like this `onClick={() => this.handleCheck(elem)}. Because 'bind' may cause "performance issue" because this function create copies from this function. Read about 'bind' under the hood. :)